src/coreclr/src/System.Private.CoreLib/src/System/Math.CoreCLR.cs

```diff
@@ -11,119 +11,176 @@
 ===========================================================*/
 
 using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
 
 namespace System
 {
     public static partial class Math
     {
+        [SuppressGCTransition]
+        [DllImport(RuntimeHelpers.QCall, CallingConvention = CallingConvention.Cdecl)]
+        private static extern double AbsInternal(double value);
         [Intrinsic]
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern double Abs(double value);
+        public static double Abs(double value) => AbsInternal(value);
 
+        [SuppressGCTransition]
+        [DllImport(RuntimeHelpers.QCall, CallingConvention = CallingConvention.Cdecl)]
+        private static extern float AbsFInternal(float value);
```

Comment:
Rather than defining `Math.AbsFInternal` and having `MathF.Abs` forward to `Math.Abs`, might it be better to do the reverse and just have all `float` variants exported from `MathF`?

I don't think it will make any actual difference; it just might be nice to normalize that oddity.

Reply:
An assert in ecall.cpp was failing: https://github.com/dotnet/runtime/pull/39474#discussion_r457456002.